public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -612,12 +612,8 @@ function sortSeriesByLabel(s1: DataFrame, s2: DataFrame): number {
 }
 
 function parseSampleValue(value: string): number {
-  switch (value) {
-    case POSITIVE_INFINITY_SAMPLE_VALUE:
-      return Number.POSITIVE_INFINITY;
-    case NEGATIVE_INFINITY_SAMPLE_VALUE:
-      return Number.NEGATIVE_INFINITY;
-    default:
-      return parseFloat(value);
+  if (INFINITY_SAMPLE_REGEX.test(value)) {
```

Comment:
is `parseSampleValue` run for every value in the response? if so, i'm wondering if there's a significant performance hit with switching to regex matching

Reply:
i have an entire comment and bench dedicated to answering this (scroll up) :rofl:

Reply:
🤦‍♂️ my bad. i attempted to quickly scan this while my toddler was busy with a book